Sources/OktaOidc/macOS/OktaRedirectServer.swift

```diff
@@ -20,14 +20,24 @@ import Foundation
 
 public class OktaRedirectServer {
 
-    var redirectHandler: OKTRedirectHTTPHandler
-    let port: UInt16
+    static let shared = OKTRedirectHTTPHandler(successURL: nil)
```

Comment:
Take a look into OktaOidcBrowserTaskMAC.deinit method. It has the following code: 
```
redirectServer?.stopListener()
```
Since your loopback server now is singleton this can lead to unpredictable behaviors. There could be cases when first signin task deallocates and calls `stopListener ()` which may affect ongoing second task

Make sure that you calling `cancelBrowserSession` before starting new authentication session. Hopefully it will help to eliminate above cases

Reply:
Yes @IldarAbdullin-okta ... @arunthotta-okta  also gave same suggestion to add `cancelBrowserSession` on client before starting new session. That took care of flow. I think we can cancel this PR for now. Sorry for the trouble.

Thanks for review @IldarAbdullin-okta nad @mikenachbaur-okta

Reply:
On Monday, I can look into this in more detail and can provide some more helpful comments.